test/LaunchDarkly.EventSource.Tests/EventSourceTests.cs

```diff
@@ -469,5 +472,23 @@ public void Exponential_backoff_should_reset_when_reconnect_count_resets()
             Assert.True(wasMessageReceivedEventRaised);
 
         }
+
+        [Fact]
+        public async Task When_error_handler_closes_event_source_no_reconnect_attempt_is_made()
```

Comment:
This test is unrelated to the error reporting changes, but verifies another piece of behavior that's important for the upcoming client changes: if an error handler calls `Close` (for instance because it saw a 401 error), we want to make sure the EventSource respects this and doesn't try to reconnect.